lib/ansible/cli/doc.py

```diff
@@ -1069,9 +1074,10 @@ def add_fields(text, fields, limit, opt_indent, return_values=False, base_indent
 
             text.append("%s%s %s" % (base_indent, opt_leadin, o))
 
+            # description is specifically formated and can either be string or list of strings
             if 'description' not in opt:
                 raise AnsibleError("All (sub-)options and return values must have a 'description' field")
-            if isinstance(opt['description'], list):
+            if isinstance(opt['description'], Sequence) and not isinstance(opt['description'], string_types):
```

Comment:
We have an `is_sequence` function which you can get from `from ansible.module_utils.common.collections import is_sequence` that handles this distinction.

Reply:
Thanks! Fixed in a1872973319a3a785c663e2e85e9838419882af5.